plots/measure.js

```diff
@@ -0,0 +1,227 @@
+    console.log('ERROR: Found output from previous run at: ', constants.OUT_PATH); // eslint-disable-line no-console
+    console.log('Please run: npm run clean'); // eslint-disable-line no-console
+    return;
+  }
+
+  const launcher = new ChromeLauncher({port: 9222, autoSelectChrome: true});
+  launcher
+    .isDebuggerReady()
+    .catch(() => launcher.run())
+    .then(() => runAnalysis())
+    .then(() => launcher.kill())
+    .catch(err => launcher.kill().then(
+        () => {
+          throw err;
+        },
+        console.error // eslint-disable-line no-console
+      ));
+}
+
+main();
+
+/**
+ * @return {!Promise}
+ */
+function runAnalysis() {
+  let promise = Promise.resolve();
+
+  // Running it n + 1 times because the first run is deliberately ignored
+  // because it has different perf characteristics from subsequent runs
+  // (e.g. DNS cache which can't be easily reset between runs)
+  for (let i = 0; i <= NUMBER_OF_RUNS; i++) {
+    // Averages out any order-dependent effects such as memory pressure
+    utils.shuffle(URLS);
+
+    const id = i.toString();
+    const isFirstRun = i === 0;
+    for (const url of URLS) {
+      promise = promise.then(() => singleRunAnalysis(url, id, {ignoreRun: isFirstRun}));
+    }
+  }
+  return promise;
+}
+
+/**
+ * @typedef {{ignoreRun: boolean}}
+ */
+let AnalysisOptions; // eslint-disable-line no-unused-vars
+
+/**
+ * @param {string} url
+ * @param {string} id
+ * @param {!AnalysisOptions} options
+ * @return {!Promise}
+ */
+function singleRunAnalysis(url, id, {ignoreRun}) {
+  console.log('Measuring site:', url, 'run:', id); // eslint-disable-line no-console
+  const parsedURL = parseURL(url);
+  const urlBasedFilename = sanitizeURL(`${parsedURL.host}-${parsedURL.pathname}`);
+  const runPath = path.resolve(constants.OUT_PATH, urlBasedFilename, id);
+  if (!ignoreRun) {
+    mkdirp.sync(runPath);
+  }
+  const outputPath = path.resolve(runPath, 'lighthouse.json');
+  const assetsPath = path.resolve(runPath, 'assets');
+  return analyzeWithLighthouse(url, outputPath, assetsPath, {ignoreRun});
+}
+
+/**
+ * @param {string} url
+ * @param {string} outputPath
+ * @param {string} assetsPath
+ * @param {!AnalysisOptions} options
+ * @return {!Promise}
+ */
+function analyzeWithLighthouse(url, outputPath, assetsPath, {ignoreRun}) {
+  const flags = {output: 'json'};
+  return lighthouse(url, flags, createConfig())
+    .then(lighthouseResults => {
+      if (ignoreRun) {
+        return;
+      }
+      return assetSaver
+        .saveAssets(lighthouseResults.artifacts, lighthouseResults.audits, assetsPath)
+        .then(() => {
+          lighthouseResults.artifacts = undefined;
+          return Printer.write(lighthouseResults, flags.output, outputPath);
+        });
+    })
+    .catch(err => console.error(err)); // eslint-disable-line no-console
+}
+
+function createConfig() {
+  const config = require('../lighthouse-core/config/perf.json');
+  config.audits = [
```

Comment:
I agree, using your own config seemed fine to me. This specific case makes sense to have a metric only config without the cruft of the other gatherers and whatnot slowing things down. @paulirish @brendankenny why use perf.json?